file_catalog/config.py

```diff
@@ -0,0 +1,51 @@
+
+from ConfigParser import SafeConfigParser
+
+import os
+import ast
+
+class Config(dict):
+    def __init__(self, path):
+        self.path = path
+        self.list_cache = {}
+
+        # read file
+        tmp = SafeConfigParser()
+        tmp.read(path)
+        self.__config_options_dict(tmp)
+
+    def __config_options_dict(self, config):
+        """
+        Parsing config file
+        Args:
+            config: Python config parser object
+        """
+        # Method copied from https://github.com/WIPACrepo/pyglidein/blob/master/client_util.py#L96
+
+        for section in config.sections():
+            self[section] = {}
+            for option in config.options(section):
+                val = config.get(section, option)
+                try:
+                    val = ast.literal_eval(val)
+                except Exception:
+                    pass
+                self[section][option] = val
+
+    def get_list(self, section, name):
+        """
+        Parses the value of a given `section`/`name` pair and returns a list.
+        It is expected that the value is a comma separated list, e.g. `a, b, c,d,e ,f , g`.
+        Note that all white spaces are removed before and after a comma.
+        """
+        if section in self.list_cache:
+            if name in self.list_cache[section]:
+                return self.list_cache[section][name]
+        else:
+            self.list_cache[section] = {}
```

Comment:
How many times are we actually executing this function?  If not often, I'd advise to just remove the cache.

Also, this is a handy decorator if you really want to use a cache:
https://wiki.python.org/moin/PythonDecoratorLibrary#Memoize
Works even better in python3 with `functools.lru_cache`